koku/api/provider/models.py

```diff
@@ -197,12 +197,187 @@ def delete(self, *args, **kwargs):
             using = router.db_for_write(self.__class__, isinstance=self)
             with schema_context(self.customer.schema_name):
                 LOG.info(f"PROVIDER {self.name} ({self.pk}) CASCADE DELETE -- SCHEMA {self.customer.schema_name}")
-                cascade_delete(self.__class__, self.__class__.objects.filter(pk=self.pk))
+                with transaction.atomic():
+                    self._cascade_delete()
+                    self._delete_from_target(
+                        {"table_schema": "public", "table_name": self._meta.db_table, "column_name": "uuid"},
+                        target_values=[self.pk],
+                    )
+                LOG.info(f"PROVIDER {self.name} ({self.pk}) CASCADE DELETE COMPLETE")
                 post_delete.send(sender=self.__class__, instance=self, using=using)
         else:
-            LOG.warning("Cannot customer link cannot be found! Using ORM delete!")
+            LOG.warning("Customer link cannot be found! Using ORM delete!")
             super().delete()
 
+    def _get_normalized_type(self):
+        """Normalize the provider type: Return lowercase type name w/o '-local'"""
+        _normalized_type = getattr(self, "_normalized_type", None)
+        if _normalized_type is None:
+            _type = self.type.lower()
+            _normalized_type = self._normalized_type = _type if "-local" not in _type else _type[: _type.index("-")]
+        return _normalized_type
```

Comment:
This function seems a bit unreadable (I don't understand why there are so many underscores). Also, instead of calling this in `cascade_delete`, can we set the `self._normalized_type` in the `delete()` func? That way, we don't need to call this function more than once and we needn't lookup whether this attribute exists or not.

```suggestion
suffix = "-local"
type = self.type.lower()
if type.endswith(suffix):
    type = type[:-len(suffix)]
return type
```

Reply:
This was done so as to not _have_ to calculate the value every time since it's static (at this point) once the record is retrieved.

Reply:
Sure, but if we just set this value in the `delete` method, this entire function would only be called once, no?